src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobReader.cs

```diff
@@ -591,9 +588,7 @@ public SignatureTypeCode ReadSignatureTypeCode()
             int length;
             if (TryReadCompressedInteger(out length))
             {
-                // Removal of trailing '\0' is a departure from the spec, but required
-                // for compatibility with legacy compilers.
-                return ReadUTF8(length).TrimEnd(s_nullCharArray);
+                return ReadUTF8(length);
```

Comment:
>return ReadUTF8(length); [](start = 16, length = 24)

Are we certain this change affects exclusively custom attributes and nothing else?

Reply:
> Are we certain this change affects exclusively custom attributes and nothing else?

Well `BlobReader.ReadSerializedString` is a public API so people can use it for all sorts of stuff; the internal usage is confined to custom attributes. The null-trimming behavior was not documented and after this change, the implementation is going to match [the docs](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.metadata.blobreader.readserializedstring?view=net-5.0) (that simply refers to ECMA-335 for the semantic).

Reply:
> Well BlobReader.ReadSerializedString is a public API so people can use it for all sorts of stuff; the internal usage is confined to custom attributes. 

Perhaps the title of the commit should have reflected the actual impact of the change then.